Service provider

deric opened the pull request:
Added support for changing service provider.

deric committed:
allow changing service provider

aerostitch commented on spec/classes/aerospike_spec.rb:
```diff
@@ -437,7 +453,10 @@
 
 				# Tests related to the aerospike::service class
-				it { is_expected.to_not contain_service('amc') }
+				it { is_expected.to contain_service('aerospike').with_hasrestart(true)\
```
Why dropping `it { is_expected.to_not contain_service('amc') }` ? It is important to keep tests that ensure that some services are not present in addition to test that some are present.

aerostitch commented on Gemfile:
```diff
@@ -4,12 +4,16 @@ group :test do
   gem "puppetlabs_spec_helper"
   gem "metadata-json-lint"
+  if RUBY_VERSION < '2.2.5'
+    gem 'ruby_dep', '< 1.4'
```
This makes the tests fail on ruby 1.9.3.

aerostitch commented on Gemfile:
```diff
@@ -4,12 +4,16 @@ group :test do
   gem "json_pure", '2.0.1' if RUBY_VERSION < '2.0'
   gem "rake"
   gem "puppet", ENV['PUPPET_VERSION'] || '~> 3.8.4'
+  gem 'hiera'
```
Why do you add hiera? It's already a dependency of the puppet gem. It seems pointless to me.

aerostitch commented on Gemfile:
```diff
@@ -21,6 +25,8 @@ group :development do
 group :development, :system_tests do
-  gem "beaker"
+  if RUBY_VERSION < '2.2.5'
+    gem 'beaker', '< 3'
```
can you use the oneliner notation instead. More compact and still really readable.

aerostitch commented on spec/classes/aerospike_spec.rb:
```diff
@@ -413,6 +413,22 @@
+        it { is_expected.to contain_class('aerospike::config').that_comes_before('Class[aerospike::service]') }
+        it { is_expected.to contain_class('aerospike::service') }
+
+        it { is_expected.to contain_service('aerospike') }
```
I don't see any test on the provider itself here. Maybe it misses a `.with_provider('systemd')` ?

aerostitch commented:
Hi @deric,

Thanks for your PR.
I added a few comments to address before merging.
Can you have a look please?

Thanks!
Joseph

deric commented:
@aerostitch Sure, hopefully I've fixed all the comments. I've squashed commits into one, to remove the mess I've created. Sorry, I was doing this in a hurry.

aerostitch commented:
Thanks a lot!

aerostitch commented:
Damn I forgot to ask you to update the doc (README.md). Can you do another PR with the doc update (just add the parameter with the default value and a short description) if you have some time please?
If not, let me know I'll take care of it.
Thanks

deric commented:
Yeah, no problem. There will be more PRs coming :)

aerostitch commented:
perfect, thanks! :)
